gpiozero/input_devices.py

```diff
@@ -34,34 +35,64 @@ class InputDevice(GPIODevice):
         The GPIO pin (in Broadcom numbering) that the device is connected to.
         If this is ``None`` a :exc:`GPIODeviceError` will be raised.
 
-    :param bool pull_up:
+    :param pull_up:
         If ``True``, the pin will be pulled high with an internal resistor. If
         ``False`` (the default), the pin will be pulled low.
+        If ``None``, the pin will be floating. As gpiozero cannot automatically
+        guess the active state when not pulling the pin, the ``active_state``
+        parameter must be passed.
+    :type pull_up: ``bool`` or ``None``
 
     :param Factory pin_factory:
         See :doc:`api_pins` for more information (this is an advanced feature
         which most users can ignore).
+
+    :param active_state:
+        If ``True``, when the hardware pin state is ``HIGH``,
+        the software pin is ``HIGH``.
+        If ``False``, the input polary is reversed:
+        when the hardware pin state is ``HIGH``,
+        the software pin state is ``LOW``.
+        Use this parameter to set the active state of the underlying pin when
+        configuring it as not pulled (``pull_up=None``).
+        When ``pull_up`` is ``True`` or ``False``, the active state is
+        automatically set to the proper value.
+
     """
-    def __init__(self, pin=None, pull_up=False, pin_factory=None):
+    def __init__(self, pin=None, pull_up=False, pin_factory=None,
+                 active_state=None):
         super(InputDevice, self).__init__(pin, pin_factory=pin_factory)
         try:
             self.pin.function = 'input'
-            pull = 'up' if pull_up else 'down'
-            if self.pin.pull != pull:
-                self.pin.pull = pull
+            if pull_up is None:
```

Comment:
The code previously did:
```python
if self.pin.pull != pull:
    self.pin.pull = pull
```

I wonder if that logic ought to be maintained, to prevent modifying the pin's pull if not necessary? :man_shrugging:

Reply:
Yup, good idea not to fiddle with pin state unless we absolutely have to. I'll add some tweaks for this.